test/e2e/apimachinery/validatingadmissionpolicy.go

```diff
@@ -355,7 +355,7 @@ var _ = SIGDescribe("ValidatingAdmissionPolicy [Privileged:ClusterAdmin]", func(
 				if err != nil {
 					return false, err
 				}
-				if policy.Status.TypeChecking != nil {
+				if policy.Status.TypeChecking != nil && len(policy.Status.TypeChecking.ExpressionWarnings) > 0 {
```

Comment:
No this won't solve the problem. The type checker will not retry when the RESTMapper does not yet recognize the requested CRD. Here, TypeChecking would be empty but non-nil, and the wait will time out.